libshpool/src/daemon/shell.rs

```diff
@@ -476,11 +442,7 @@ impl SessionInner {
                     }
                 }
 
-                if !matches!(args.session_restore_mode, config::SessionRestoreMode::Simple) {
-                    if let (Some(s), true) = (output_spool.as_mut(), has_seen_prompt_sentinel) {
-                        s.process(buf);
-                    }
-                }
+                output_spool.process(buf);
```

Comment:
Are we still respecting the prompt sentinel? That seems like something that should happen outside the scope of the output spool since it is part of the shell setup protocol, not really the in-mem terminal.

Reply:
Ah indeed. Added the missing condition to check for `has_seen_prompt_sentinel`.